docs/user_guide/command_parser.md

```diff
@@ -0,0 +1,233 @@
+              value: "{{ item.mtu.matches.0 }}"
+            - key: description
+              value: "{{ item.description.matches.0 }}"
+  loop: "{{ interfaces }}"
+  export: yes
+  register: interface_facts
+
+```
+
+`parser_templates/ios/show_version.yaml`
+
+```yaml
+
+---
+- name: parser meta data
+  parser_metadata:
+    version: 1.0
+    command: show version
+    network_os: ios
+
+- name: match version
+  pattern_match:
+    regex: "Version (\\S+),"
+  register: version
+
+- name: match model
+  pattern_match:
+    regex: "^Cisco (.+) \\(revision"
+  register: model
+
+- name: match image
+  pattern_match:
+    regex: "^System image file is (\\S+)"
+  register: image
+
+- name: match uptime
+  pattern_match:
+    regex: "uptime is (.+)"
+  register: uptime
+
+- name: match total memory
+  pattern_match:
+    regex: "with (\\S+)/(\\w*) bytes of memory"
+  register: total_mem
+
+- name: match free memory
+  pattern_match:
+    regex: "with \\w*/(\\S+) bytes of memory"
+  register: free_mem
+
+- name: export system facts to playbook
+  set_vars:
+    model: "{{ model.matches.0 }}"
+    image_file: "{{ image.matches.0 }}"
+    uptime: "{{ uptime.matches.0 }}"
+    version: "{{ version.matches.0 }}"
+    memory:
+      total: "{{ total_mem.matches.0 }}"
+      free: "{{ free_mem.matches.0 }}"
+  export: yes
+  register: system_facts
+
+```
+
+## Sample Playbooks
+
+To extract the data defined in your file, create a playbook that includes the Network Engine role and references the `content` and `file` parameters of the `command_parser` module. 
+
+Each example playbook below runs a show command, imports the Network Engine role, extracts data defined from the text output of the command by matching it against the rules defined
+in your data definition file, and views the results. (The last step is for demonstration purposes only.) 
+
+Make sure the `hosts` definition in the playbook matches a host group in your inventory - in these examples, the playbook expects a group called `ios`.
+
+The first example parses the output of the `show interfaces` command on IOS and creates facts from that output:
+
+```yaml
+
+---
+
+# ~/my-playbooks/gather-interface-info.yml
+
+- hosts: ios
+  connection: network_cli
+
+  tasks:
+  - name: Collect interface information from device
+    ios_command:
+      commands:
+        - show interfaces
+    register: ios_interface_output
+
+# the command output being parsed is stored in the stdout of ios_version_output
+# you can see a formatted version if you uncomment these three lines:
+#  - name: debug
+#    debug:
+#      var: ios_interface_output['stdout_lines']
+
+  - name: import the network-engine role
+    import_role:
+      name: ansible-network.network-engine
+
+  - name: Generate interface facts as JSON
+    command_parser:
+      file: "parser_templates/ios/show_interfaces.yaml"
+      content: "{{ ios_interface_output.stdout.0 }}"
+
+  - name: Display interface facts in JSON
+    debug:
```

Comment:
No need to use `debug` as displaying interface facts is on `export` and `register` directive in command_parser module.
When `export` is set to `true` in parser file, the facts will be displayed anyway.

Reply:
I added the debug statement so I could see the contents, because on my first few runs I had a lot of tasks that "succeeded" but the content of the variables was all "NULL". It is incredibly helpful for debugging, especially because it demonstrates the difference between `ios_interface_output.stdout.0` and `interface_facts`. The latter is defined in the parser template, and the example debugging task closes that loop. I'll play with different ways to include it - maybe commented out like the example further up?

Reply:
@acozine I can simply get `interface_facts` on display without having `debug` statement. You just need to use export in your parser template parser_templates/ios/show_interfaces.yaml file. that is how the directive is designed.
```
ok: [localhost] => {
    "ansible_facts": {
        "interface_facts": [
            {
                "GigabitEthernet0/0": {
                    "config": {
                        "description": "OOB Management", 
                        "mtu": 1500, 
                        "name": "GigabitEthernet0/0", 
                        "type": "iGbE"
                    }
                }
            }, 
            {
                "GigabitEthernet0/1": {
                    "config": {
                        "description": "test-interface", 
                        "mtu": 2000, 
                        "name": "GigabitEthernet0/1", 
                        "type": "iGbE"
                    }
                }
            }, 
            {
                "GigabitEthernet0/2": {
                    "config": {
                        "description": "test-interface-2", 
                        "mtu": 2000, 
                        "name": "GigabitEthernet0/2", 
                        "type": "iGbE"
                    }
                }
            }
        ]
    }, 
    "changed": false, 
    "included": [
        "/home/tguha/RHWork/network-engine/tests/text_parser/parsers/ios/show_interfaces.yaml"
    ]
}

```
Same for `textfsm`, since you are using `name: interface_facts`. `name` itself is responsible for displaying the facts on display. no need of `debug`. 
```
TASK [textfsm : show_version test] ********************************************************************************************************************************************************************************
task path: /home/tguha/RHWork/network-engine/tests/textfsm/tasks/ios.yaml:27
File lookup using /home/tguha/RHWork/network-engine/tests/textfsm/output/ios/show_version.txt as file
ok: [localhost] => {
    "ansible_facts": {
        "system_facts": [
            {
                "image": "\"flash0:/vios-adventerprisek9-m\"", 
                "model": "IOSv", 
                "uptime": "10 weeks, 6 days, 22 hours, 30 minutes", 
                "version": "15.6(2)T"
            }
        ]
    }, 
    "changed": false
}
```
That is how the module is designed. Module doc has the same info.

Reply:
@acozine Not sure if I understand your purpose of using `debug` here, if you are thinking to have an example with debug to show how to access the facts, then something similar to the following would make sense.
```
- name: Access interface facts
  debug:
    var: interface_facts[0]['GigabitEthernet0/0']['config']['description']
```
Also add a note in comment that this `interface_facts` is the registered variable used in parser_template.
`Access interface facts` task name indicates that the purpose of this task is to show how one can access the facts rather than use it for display purpose.
For displaying facts we should encourage user to use `export` in parser template.

Same applicable for textfsm_parser.